numba/core/boxing.py

```diff
@@ -73,16 +73,26 @@ def box_float(typ, val, c):
 
 @unbox(types.Float)
 def unbox_float(typ, obj, c):
-    fobj = c.pyapi.number_float(obj)
-    dbval = c.pyapi.float_as_double(fobj)
```

Comment:
This line could end up calling `PyFloat_asDouble` with `NULL`, which is not allowed.